### PR TITLE
TransGAN: A Transformer based GAN architecture

### LINKED ISSUE #1610: TransGAN: A Transformer based GAN architecture
### Introduction paragraph(s):

Can we build a strong Generative Adversarial Network (GAN) utterly free of convolutions?

This article aims to discuss this recent [paper](https://arxiv.org/pdf/2102.07074.pdf), a buzz in the AI community over the past couple of weeks. The paper aims to re-design the architecture for Generative Adversarial Networks (GANs) using only pure transformer-based architecture.

### Key takeaways: 

What are the 3-5 most important things the reader should understand or be able to do after reading this article? 

1. What is a Generative Adversarial Network (GAN)?
2. A description of the TransGAN architecture
3. Tricks for TransGAN

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

1. [TransGAN: Two Transformers Can Make One Strong GAN](https://arxiv.org/pdf/2102.07074.pdf)

### PR DESCRIPTION
**Pull request by Bayler**
This Fixes #1610

### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you placed your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/transgan-a-transformer-based-gan-architecture/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+There are three main techniques employed to train TransGAN better. They include:
+
+- Data augmentation
+- Localized initialization for self-attention
+- Multi-task co-training for generator with self-supervised auxiliary loss
+
+### Wrapping Up
+
+In this article, we have discussed the TransGAN paper. It attempts to re-design GANs architecture by replacing convolutions with pure transformers.
+Over the past years, we've seen a shift in the use of CNNs to use powerful transformers to perform image classification tasks. Transformers are now being attempted on image generation tasks.  Most of these proposed models still have some challenges. One case is in the use of transformers for image generation tasks such as in GANs. But, they still appear to be very promising architectures.
```
Comment (collins-a) [inline articles/transgan-a-transformer-based-gan-architecture/index.md:128]:
```suggestion
Over time, we have seen a shift in the use of CNNs to use powerful transformers for image classification tasks. Transformers are now being tried on image generation tasks. However, most of these proposed models still pose some challenges. An instance is the use of transformers for image generation tasks, such as in GANs. But, they still appear to be very promising architectures.
```

File: articles/transgan-a-transformer-based-gan-architecture/index.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+**2. Co-training with Self-Supervised Auxiliary Task**
+
+![Co-training with Self-Supervised Auxiliary Task](/engineering-education/transgan-a-transformer-based-gan-architecture/co-training.PNG)
+
+*[Image Source: Arxiv](https://arxiv.org/pdf/2102.07074.pdf)*
+
+This is a super-resolution task. It is a task performed besides the whole GAN training. 
+
+The GAN training involves the discriminator model receiving images from both the generator and the real dataset. Through training, this gives us the primary GAN loss.
```
Comment (collins-a) [inline articles/transgan-a-transformer-based-gan-architecture/index.md:93]:
```suggestion
The GAN training involves the discriminator model receiving images from both the generator and the real dataset. Through training, we get the primary GAN loss.
```

**Comment by collins-a** [comment]
@Bayler good dissection of transformer-based GAN architectures. You did well directing the reader to additional reading material while attempting to keep the content as simple as possible since it's a rather complex topic. I believe we're ready for the final review

**Comment by hectorkambow** [comment]
plagiairsm check done

**Pull request merged**